finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
AFAIU, (Http)ServerDispatcher#[handle](https://github.com/twitter/finagle/blob/finagle-6.35.0/finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala#L63) is called after all the filters are applied? In `handle()`, we add a Content-Length header field with a value of 0 in [here](https://github.com/twitter/finagle/pull/521/files#diff-225f97996577fc88c90976a35c713e4cR102). If I understand finagle's behaviour correctly, I think we can't avoid putting the code there?
(but I have to admit this is not nice though :(

I might be wrong, please give me your idea :)